### DISCUSSION TITLE
Unable to sign the AIR file when publishing Mac app in Adobe Animate

### DISCUSSION THREAD

**Discussion by scott-higgs** [airsdk/Adobe-Runtime-Support#1968] [Q&A]
Seems to me in the past this was caused by absent Apple root certificates, so I downloaded current certificates, created a new Apple Distribution certificate (which is what I've used in the past).  I've tried creating the certificate through the Apple Developer website, and in Xcode.  I'm exporting the key as .p12 and selecting in the AIR settings Signature tab, per usual.  I've tried 33.1.1.856 and 32.0.0.116.  Not sure what I'm missing, but I can't get past the following error in the publish:

Could not sign the AIR file.
Unable to build a valid certificate chain for the signer.

Any thoughts?

**Comment by ajwfrost** [comment]
If you're using a recent AIR SDK at this point, can you please edit the "adt.cfg" file under the lib folder, and un-comment "DebugOut=true". Then try it again, and it should generate a file for you "~/adt.log" i.e. in the parent folder of Documents, Downloads etc, /Users/username, there should just be a new file adt.log. This should have some details in there of the search it's doing for the path from your certificate to a root certificate... and may include some other clues as to the problem.

thanks

**Reply by scott-higgs** [reply]
The only pertinent line in the resulting log file is:
General security error: Unable to build a valid certificate chain for the signer.
Which doesn't help overmuch. On the other hand, I've been wishing there was a way to view the adt log files in the AIR sdk for years, so thanks _very much_ for that!

**Reply by scott-higgs** [reply]
Sorry, not the only pertinent line.  Missed this bit:
-storetype PKCS12 -keystore [correct path to my .p12] -storepass xxxxxx
But that all looks correct to me.

**Reply by ajwfrost** [reply]
Hmm... so there are two options for code-signing, but I would expect to see one of these lines in the adt.log file:
```
Codesign request for IPA: codesign --force --sign  ....
```
if it's using native code-signing (from the OS), or
```
Checking certificate chain for the issuer
```
if it's using the Java code in ADT itself.

I can see where the security error there is happening but it's not clear why this would happen - it's trying to verify the certificate and I think that involves checking the trust path.

From that adt.log file, you should see the command-line that Animate is using to call into ADT: is it possible to open up a terminal window on the macOS machine (not any remote-access via ssh/similar, but directly on then machine itself) and try entering the same command-line there? I'm wondering if there's a permissions block that's happening because it's being launched via Animate... (plus, what mac OS version and hardware type do you have?)

thanks

**Reply by scott-higgs** [reply]
Thanks @ajwfrost .  When I run the generated command-line in terminal, the build fails with a single line of feedback:

Unable to build a valid certificate chain for the signer.

MacBook Pro (16-inch, 2021)
MacOS Monterrey 12.4
Apple M1 Max
